haigha/classes/basic_class.py

```diff
@@ -53,8 +54,30 @@ def _cleanup(self):
         self._get_cb = None
         self._recover_cb = None
         self._cancel_cb = None
+        self._return_listener = None
```

Comment:
I'd like to see this use the existing naming scheme, so this and the corresponding set method would be `return_cb` instead of `return_listener`